app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -361,6 +361,9 @@ def remove_content
     param :id, :number, :required => true, :desc => N_("repository ID")
     param :content, File, :required => true, :desc => N_("Content files to upload. Can be a single file or array of files.")
     param :content_type, RepositoryTypeManager.uploadable_content_types(false).map(&:label), :required => false, :desc => N_("content type ('deb', 'docker_manifest', 'file', 'ostree', 'rpm', 'srpm')")
+    param :ostree_repository_name, String, :desc => N_("Name of repository in OSTree archive")
+    param :ostree_ref, String, :desc => N_("OSTree ref branch that holds the reference to the last commit")
+    param :ostree_parent_commit, String, :desc => N_("Checksum of a parent commit with which the OSTreee content needs to be associated")
```

Comment:
i wonder if we could add support for this in generic content type registry?

Reply:
I could envision the repository type registry supporting this via something like this:
~~~
  generic_content_type 'ostree_ref',
                       model_class: Katello::GenericContentUnit,
                       pulp3_api: PulpOstreeClient::ContentRefsApi,
                       pulp3_service_class: Katello::Pulp3::GenericContentUnit,
                       model_name: lambda { |pulp_unit| pulp_unit["name"] },
                       model_version: lambda { |pulp_unit| pulp_unit["version"] },
                       import_attribute :parent_commit, :api_param => :ostree_parent_commit, :description => "Checksum of a parent commit blahblahl", :type => String, :required => false
                       import_attribute :repository_name, :api_param => :ostree_repository_name, :description => "Name of the repository in the ostree archive", :type => String, :required => true


~~~

and then these options could be rendered using the repository type manager, and could be fetched from the params and injected dynamically based on content type.